spec/service_consumers/pact_helper.rb

```diff
@@ -44,4 +46,43 @@ def url_encode(str)
       create(:place, service_slug: service.slug, latitude: 50.742754933617285, longitude: -1.9552618901330387)
     end
   end
+
+  provider_state "a place exists with a postcode and areas" do
+    set_up do
+      service = create(:service, slug: "local-authority")
+      create(:place, service_slug: service.slug, postcode: "WC2B 6SE")
+
+      areas = [
+        { "name" => "Westminster City Council", "type" => "LBO" },
+        { "name" => "London", "type" => "EUR" },
+      ]
+
+      response = {
+        "wgs84_lat" => 51.516,
+        "wgs84_lon" => -0.121,
+        "postcode" => "WC2B 6SE",
+      }
+
+      area_response = Hash[areas.map.with_index do |area, i|
+        [i,
+         {
+           "codes" => {
+             "ons" => area["ons"],
+             "gss" => area["gss"],
+             "govuk_slug" => area["govuk_slug"],
+           },
+           "name" => area["name"],
+           "type" => area["type"],
+           "country_name" => area["country_name"],
+         }]
+      end]
```

Comment:
I was surprised to see the closing `]` here - it's not very intuitive to see what's going on! Looks like `area_response` ends up with this structure:

```ruby
 {0=>{"codes"=>{"ons"=>nil, "gss"=>nil, "govuk_slug"=>nil}, "name"=>"Westminster City Council", "type"=>"LBO", "country_name"=>nil}, 1=>{"codes"=>{"ons"=>nil, "gss"=>nil, "govuk_slug"=>nil}, "name"=>"London", "type"=>"EUR", "country_name"=>nil}}
```

I think it's fine not to aim for clever code here, and just have one big hash, even if there's some repetition (we can refactor later if we want):

```ruby
  area_response = {
    "wgs84_lat" => 51.516,
    "wgs84_lon" => -0.121,
    "postcode" => "WC2B 6SE",
    "results" => [
        {
          "name" => "Westminster City Council",
          "type" => "LBO",
           "codes" => {
             "ons" => ...,
             "gss" => ...,
             "govuk_slug" => ...,
           },
           "country_name" => "England",
        },
        {
          "name" => "London",
          "type" => "EUR",
           "codes" => {
             "ons" => ...,
             "gss" => ...,
             "govuk_slug" => ...,
           },
           "country_name" => "England",
        },
    ],
  }
```

Reply:
👍